File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            match i:
                case '(':
                    stack.append(')')
                case '{':
                    stack.append('}')
                case '[':
                    stack.append(']')
                case default:
                    if not stack or stack.pop() != i: return False
                    
        return len(stack) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    for(let i=0; i<s.length; i++){
        let curVal = s[i];
        
        switch(curVal){
                 case '(':
                stack.push(')');
                break;
            case '{':
                stack.push('}');
                break;
                    case '[':
                        stack.push(']');
                        break;
                    default:
                        if(stack.pop() !== curVal) return false;
                        break;
                    
        }
           
        
    }
        
        return stack.length === 0
};